apollo-router/src/plugins/telemetry/config_new/mod.rs

```diff
@@ -27,9 +25,9 @@ pub(crate) mod spans;
 pub(crate) trait Selectors {
     type Request;
     type Response;
-    fn on_request(&self, request: &Self::Request) -> LinkedList<KeyValue>;
-    fn on_response(&self, response: &Self::Response) -> LinkedList<KeyValue>;
-    fn on_error(&self, error: &BoxError) -> LinkedList<KeyValue>;
+    fn on_request(&self, request: &Self::Request) -> Vec<KeyValue>;
```

Comment:
I suspect that these could return iterators (impl traits in traits should work now) instead of allocating lots of vectors, since they're almost always consumed as iterators right away, but I'm not sure there will be a huge benefit in doing that

Reply:
I tried using Iterator but at the end it won't help because we often have to manipulate vec to mutate and insert attributes in that vec and then convert it again to an iterator